src/Config/ConfigOption.php

```diff
@@ -0,0 +1,290 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Config;
+
+use Entity;
+
+/**
+ * A configuration option
+ */
+class ConfigOption
+{
+    /**
+     * @param ConfigScope[] $scopes The valid scopes
+     * @param ConfigSection $section The configuration section this option belongs to
+     * @param string $name The internal name of the option
+     * @param string $label The label of the option
+     * @param ConfigType $type The type of the option
+     * @param array $type_options The options for the type of the option
+     * @param string $context The context of the option
+     */
+    public function __construct(
+        private readonly array $scopes,
+        private readonly ConfigSection $section,
+        private readonly string $name,
+        private readonly string $label,
+        private readonly ConfigType $type,
+        private readonly array $type_options = [],
```

Comment:
`$type_options` corresponds to the options of the macro that will be used to render input corresponding to the `ConfigOption` instance. It may be hard to know what are the valid options without doing some reverse engineering.

Instead of having a `ConfigType` enum, we could have dedicated classes for each type, each one having its own parameters, and its own rendering method. It would probably be easier for plugin developers to register new specific types, and it would be probably easier to maintain that `if/else` or `switch/case` at many places.